### ISSUE TITLE
[HOLD for payment 2023-10-25] [$1000] "This is the beginning of ..." message shows up on LHN of a new chat for a deleted thread instead of [message deleted]

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#19722]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Post a new message in a new chat with no history (i.e. PARENT MESSAGE) and reply in thread (i.e. CHILD MESSAGE)
2. Go back to the chat and delete the new message (i.e. PARENT MESSAGE)
3. Notice the LHN does not say [message deleted], even though the chat states this

![image](https://github.com/Expensify/App/assets/30815269/555671f9-8c32-425b-93cb-c0d1e486c2ba)

## Expected Result:

In the LHN for a PARENT MESSAGE, [message deleted] should show instead of "This is the beginning of..." for

## Actual Result:

In the LHN for a new chat with a deleted PARENT MESSAGE, we see "This is the beginning of..."

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.19.1

**Reproducible in staging?:** yes

**Reproducible in production?:** yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/264821e0-cf8c-4041-a665-9ecdc08904ad

https://github.com/Expensify/App/assets/93399543/0383d80f-3a43-4f88-82be-70b824463753

**Expensify/Expensify Issue URL:**

**Issue reported by:** @Nathan-Mulugeta

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1684758578526459

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~018360917605fa9833</li>
        <li>Upwork Job ID: 1663327696563245056</li>
        <li>Last Price Increase: 2023-09-22</li>
<li>Automatic offers: </li>
<ul>
<li>rojiphil | Contributor | 26910081</li>
<li>Nathan-Mulugeta | Reporter | 26910082</li>
</ul></li>
    </ul>
</details>

**Comment by jliexpensify** [comment]
I can repro this on v1.3.18-2 (prod) and Staging.

On a chat with history, the deleted thread doesn't show in the LHN:

![image](https://github.com/Expensify/App/assets/30815269/9159b962-9841-4ca9-9987-4a5a2433a9a4)

On a new chat without history, I see `___fake___` briefly, then the "This is the beginning" message:

![Uploading image.png…]()

**Comment by jliexpensify** [comment]
Confirming what message should show: https://expensify.slack.com/archives/C049HHMV9SM/p1685235997879309?thread_ts=1684758578.526459&cid=C049HHMV9SM

**Comment by jliexpensify** [comment]
@rushatgabhane @puneetlath - it looks like another contributor has [proposed the solution to this issue on another GH](https://github.com/Expensify/App/issues/19629#issuecomment-1566447331)

**Comment by jliexpensify** [comment]
If the solution in the other GH is the same/fixes this issue, then I think we should close this out and pay @Nathan-Mulugeta for reporting.

**Comment by puneetlath** [comment]
I think maybe this should be a back-end change? 

In any case, I'm going to assign the CME role over to chirag/brandon since this relates to threads.

**Comment by jliexpensify** [comment]
Not overdue, waiting on @stitesExpensify and @chiragsalian

**Comment by chiragsalian** [comment]
I actually think we should do nothing here. Lets take a step back to define things. In the LHN, for a single chat we see two rows
First row - Email or Name of the person we are DM-ing (or room name for policies)
Second row - Last unread non-deleted message

Because in the chat, the message is deleted, we have decided not to show it in the second row of the LHN so this is a design choice and by that choice it's correct that we show "This is the beginning of...". 

I don't think we need to do anything here. Feel free to correct me if I'm missing something.

**Comment by jliexpensify** [comment]
Toa dd to Chirag's above comment, there are some additional clarifying questions/context in the [thread](https://expensify.slack.com/archives/C049HHMV9SM/p1686005522092739?thread_ts=1684758578.526459&cid=C049HHMV9SM) - waiting on a response before we decide what to do.

**Comment by jliexpensify** [comment]
Still discussing behaviours in Slack thread, should be close to wrapping it up soon.

**Comment by jliexpensify** [comment]
@chiragsalian has weighed in [here](https://expensify.slack.com/archives/C049HHMV9SM/p1686272128101069?thread_ts=1684758578.526459&cid=C049HHMV9SM) and considers this current issue to not be a bug.

However, this issue could be tweaked if his proposed change is accepted, and we can source proposals. I'll update the issue once we have consensus.

**Comment by jliexpensify** [comment]
bump @chiragsalian on this [comment](https://github.com/Expensify/App/issues/19722#issuecomment-1588297916)

**Comment by jliexpensify** [comment]
Not overdue!

**Comment by chiragsalian** [comment]
@jliexpensify, yes correct.

In past for LHN chat second row we would show "last non-deleted message". Hence we didn't show `[Deleted message]` in the UI.

Moving forward for LHN chat second row let's show "last visible message". So now we're ready to accept proposals such that `[Deleted message]` is shown in the LHN chat second row.

Looks like there are already some proposals, @rushatgabhane will you be able to review them when you get the chance? Feel free to let us know if you have any questions 🙂

**Comment by jliexpensify** [comment]
Awesome, thanks @chiragsalian ! Ready to assign this one once we make a decision.

**Comment by jliexpensify** [comment]
Bump @chiragsalian to the above comment.

**Comment by chiragsalian** [comment]
> okayy but we should also fix the lastMessageText returned by the backend, no?

Ah you are correct, but backend cannot return `[Deleted Message]` since that's translated on the front end.
Which means we most likely have to create a new key, I'm thinking `isLastMessageParentAction`. If it's true and `lastMessageText` is empty then we show the translated deleted message. Thoughts?

**Comment by jliexpensify** [comment]
@rushatgabhane @chiragsalian should we move this back to External and source new proposals?

**Comment by jliexpensify** [comment]
Bump @rushatgabhane for a review!

**Comment by chiragsalian** [comment]
Sounds good, i created a [backend issue](https://github.com/Expensify/App/issues/21963) since i would like to track it separately.

Updated issue title to mention the HOLD.

**Comment by DanutGavrus** [comment]
Hi, I see this is on hold for the BE issue. But I'm curious if this would work before the BE issue:

## Proposal

### Please re-state the problem that we are trying to solve in this issue.
"This is the beginning of ..." message shows up on LHN of a new chat for a deleted thread instead of [message deleted]

### What is the root cause of that problem?
1. In `SidebarUtils` -> `getOptionData` we have `const lastMessageTextFromReport = OptionsListUtils.getLastMessageTextForReport(report)`;
2. In `OptionsListUtils` -> `getLastMessageTextForReport` we do not check if the `lastReportAction` is a *deleted action* and the "bit ugly" code sets the `lastMessageTextFromReport` to the latest visible action that is not a whisper:
https://github.com/Expensify/App/blob/6a7c1b2bb3d938b0b2c6388370a818f31e58b93f/src/libs/OptionsListUtils.js#L379-L403

### What changes do you think we should make in order to solve the problem?
Either **I.** Add a check after the last `else` such as:
```js
if (lastMessageTextFromReport != 'Reply' && ReportActionUtils.isDeletedAction(lastReportAction)) {
    lastMessageTextFromReport = Localize.translateLocal('parentReportAction.deletedMessage');
}
```
`Reply` comes from the BE if the original thread message was deleted. With this option, we do **not** replace the `Reply` text in LHN, but for Spanish, we'll keep seeing the text in English:
![image](https://github.com/Expensify/App/assets/56603839/fd0fa09d-f3e9-46ed-99c6-6f75066adf18)

Or **II.** Add a check for `isDeletedAction` before the last `else`(and maybe also replace the if elses with a switch):
```js
else if (ReportActionUtils.isDeletedAction(lastReportAction)) {
    lastMessageTextFromReport = Localize.translateLocal('parentReportAction.deletedMessage');
} else {
```
With this option, this issue seems fixed & we also replace the `Reply` text with localized `[Deleted message]` text.

Is something wrong with this, did I miss any test cases? Thanks!

**Comment by chiragsalian** [comment]
Okay, I'm done with the backend changes.

Slight twist, we need to do the App PR first before the backend change is live because if the backend goes first it would return lastMessageText and lastMessageHtml as empty for any deleted action instead of the previous message which is what we currently do. 

So here's how the backend changes work,
1. We return the last action text/html, whether it's deleted or not PHP layer will return it. So onyx lastMessageHtml and lastMessageText will be empty if the last action was deleted. Previously this would be the most recent non-deleted action.
2. Backend now returns a new onyx data point i.e., `isLastMessageDeletedParentAction`. I changed the key name since this more closely matches report actions where we have `isDeletedParentAction` for reportActions.
3. Backend pushes onyx data for this key `isLastMessageDeletedParentAction` when we call OpenReport (to fetch the latest value), when we add a comment (isLastMessageDeletedParentAction is set to null since we cannot add a deleted comment), and when we delete a comment.

So next steps,
1. I think the front end code should be deployed first to account for empty lastMessageHtml/lastMessageText along with accounting for `isLastMessageDeletedParentAction` to display the proper LHN content. This way the code is functionally safe once the backend changes do go live.
2. Front end should also optimistically set `isLastMessageDeletedParentAction` by setting it to true on parent report when latest parent action is deleted and clear it when a new comment is added.

So we'll need to accept proposals that accounts for these changes. Does this make sense @rushatgabhane. Let me know if anyone has any questions. I can place the backend code for review but I'll put it on HOLD until App code is live so that it does not disrupt users.

**Comment by rushatgabhane** [comment]
> It will return lastMessageHtml: '' and isLastMessageDeletedParentAction : true only if the deleted comment has a child thread, else it will returns the last non deleted action and isLastMessageDeletedParentAction will be null.

sounds good.

we can work with @dukenv0307's [proposal](https://github.com/Expensify/App/issues/19722#issuecomment-1569838965) now

**Comment by chiragsalian** [comment]
Cool, feel free to create the PR @dukenv0307.

**Comment by jliexpensify** [comment]
Gah Upworks joib is closed, frustrating!

Here's a new manual job: https://www.upwork.com/jobs/~01728e46a66566b0d2

**Comment by jliexpensify** [comment]
Hired Nathan M for reporting. @dukenv0307 can you please edit your GH handle to include your full name in Upworks and apply here? https://www.upwork.com/jobs/~01728e46a66566b0d2

@rushatgabhane not going to hire you, please record your own C+ payment :)

**Comment by jliexpensify** [comment]
Hired you @dukenv0307 ! So Chirag just went on paternity leave, so I'm going to re-apply the Engineering label to have someone else take this over the line.

**Comment by jliexpensify** [comment]
Hi @flodnv - just assigning an engineer as Chirag is off now. A PR has just been submitted so nothing needed from you yet - cheers!

**Comment by flodnv** [comment]
Okay 👍 requested a review from @rushatgabhane https://github.com/Expensify/App/pull/22864#event-9835367325

**Comment by jliexpensify** [comment]
Not overdue, PR is being worked on!

**Comment by jliexpensify** [comment]
Not overdue

**Comment by s-alves10** [comment]
I'm not sure why we need to add `isLastMessageDeletedParentAction`.
As @dukenv0307 said, we have many regressions due to the introduction of this. And we get this value from the report actions.

I think there is no need to add this for sure.
Let me know your thoughts

**Comment by stitesExpensify** [comment]
FYI Chirag will be OOO for the next couple of months. @grgia do you want to take on the backend changes here? If not we can probably open this up to general engineering

**Comment by chiragsalian** [comment]
@stitesExpensify, the backend is done here,
https://github.com/Expensify/Web-Expensify/pull/38034
https://github.com/Expensify/Auth/pull/8221

Can @stitesExpensify or @grgia take over it? Mostly addressing merge conflict and testing it out.

Its just on HOLD because if the backend is merged before App is live then deleting any thread message will show "This is the beginning of.." in LHN for the thread since it will no longer show last non-deleted message. Hence App PR should go live before the BE change. 

> I'm not sure why we need to add isLastMessageDeletedParentAction.
> And we get this value from the report actions.
> I think there is no need to add this for sure.

Its a valid thought. I could be mistaken but I'm pretty sure we need it. Here are my reasons why i think we need `isLastMessageDeletedParentAction`.
1. Users load all onyx report data first. They don't always fetch every report actions for every report. If a user has a thread in the LHN with it latest comment deleted and the user has not fetched the report action. So relying on report action for LHN UI is not ideal.
2. So the next question becomes why don't we use the existing lastMessageHtml/lastMessageText instead of the new key `isLastMessageDeletedParentAction`. Its because `[Message Deleted]` is a system comment i.e., its specific to the translation user has chosen. Hence backend cannot return a specific message but should instead return a key like `isLastMessageDeletedParentAction` so that the app can translate the message into the users specific language.

If you still feel like this is not needed could you explain how else we could solve this? 🙂

**Comment by marcaaron** [comment]
Just a heads up that the PR linked here caused [a regression that blocked the deploy](https://github.com/Expensify/App/issues/23401). We put up a fix and CP'd it to staging, but I have been told by @s-alves10 that it will re-introduce the original issue here. So, I think we can consider it a "soft" revert and the work we set out to do in this issue is not acceptable. 

Let's find some way to fix this without introducing any regressions? Thanks!

**Comment by s-alves10** [comment]
@chiragsalian 

Yes. Fetching all report actions in advance to show LHN is not a good choice because it would take some time and traffic. But you know, we have `lastMessageText` and `lastMessageTranslationKey` for this purpose. You said `[Deleted message]` is a system comment. We can show system message using `lastMessageTranslationKey`, right?

I think `isLastMessageDeletedParentAction` isn't needed at all. Instead, we have to fix the issues regarding the deletion of the action.

There are several issues now
https://github.com/Expensify/App/issues/19722
https://github.com/Expensify/App/issues/19722

**Comment by rojiphil** [comment]
@marcaaron  Regarding #23445, the issue is not resolved completely. Actually, as I understand, there are two ways in which we can delete the Parent Message comment. a) From Thread Report and b) From Parent Report. Currently, (a) will work but (b) will fail because we are using `ReportUtils.isThreadFirstChat`. `isThreadFirstChat` checks whether the `reportID` and the `childReportID` are same. However, in the case of (b), `reportID` and `childReportID` will be different. We can solve this by using `reportAction.childReportID` and `ReportUtils.isThreadParent` as I had mentioned in my original proposal. However, as you mentioned, we can avoid `lastReadTime` and use a dummy one `updateReportInLHN` as added already. Here is the proposed code:

```
    // Check to see if the report action we are deleting is a thread parent action. In this case, we need to trigger
    // an update to let the LHN know that the parentReportAction is now deleted.
    if(ReportUtils.isThreadParent(reportAction)) {
        optimisticData.push({
            onyxMethod: Onyx.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT}${reportAction.childReportID}`,
            value: {updateReportInLHN: true},
        });
    }
```

**Comment by rojiphil** [comment]
## Proposal

Have [proposed an alternate solution here](https://github.com/Expensify/App/issues/23450#issuecomment-1658337106) for this issue. Adding a reference here.

**Comment by marcaaron** [comment]
@s-alves10 @rojiphil Heads up that I am not actively managing or assigned to this issue. Please tag the original participants or start a conversation in Slack if you need help. Thank you.

**Comment by rojiphil** [comment]
## Proposal

[Updated proposal](https://github.com/Expensify/App/issues/23450#issuecomment-1658337106) to include test video.  Looks pretty smooth to me. 
@rushatgabhane @flodnv @chiragsalian  Kindly review. 
If this needs to be fixed here, I can move the proposal here too. Thanks.

**Comment by jliexpensify** [comment]
Based off this [comment](https://github.com/Expensify/App/issues/19722#issuecomment-1648818264), I'm going to remove the HOLD FOR PAYMENT in the title to avoid confusion.

**Comment by flodnv** [comment]
@rushatgabhane please have a look and let me know what you think

**Comment by rojiphil** [comment]
[Updated proposal](https://github.com/Expensify/App/issues/23450#issuecomment-1658337106)
@rushatgabhane Have updated the proposal with additional points (6) and (7). Also, have updated videos.
Looking forward to your review comments. Thanks.

**Comment by rojiphil** [comment]
## Proposal

[Updated proposal](https://github.com/Expensify/App/issues/23450#issuecomment-1658337106)
@rushatgabhane Have updated the proposal with additional points (6) and (7). Also, have updated videos.
Looking forward to your review comments. Thanks.

**Comment by jliexpensify** [comment]
Bump @rushatgabhane !

**Comment by s-alves10** [comment]
@rushatgabhane @flodnv 
Please check my comments [here](https://github.com/Expensify/App/issues/19722#issuecomment-1649225218)

**Comment by jliexpensify** [comment]
Heads up - Florent is OOO. @rushatgabhane can you handle the comments?

**Comment by jliexpensify** [comment]
@rushatgabhane bumping you to address comments

**Comment by rushatgabhane** [comment]
Could someone please help me understand what is the regression 

https://github.com/Expensify/App/issues/23450 is not reproducible.

https://github.com/Expensify/App/assets/29673073/4fd8e200-0a04-4b5c-b285-2f0330fbcd83

**Comment by rojiphil** [comment]
> #23450 is not reproducible.
 
@rushatgabhane 

I also tried reproducing the specific issue of #23450 but could not. Did not dwell deeper but looks like [this PR](https://github.com/Expensify/App/pull/24409) or some other implementation may have fixed this. My proposal addressed the same with an alternate implementation as mentioned in [point(3) of the proposal](https://github.com/Expensify/App/issues/23450#issuecomment-1658337106). 

But, the issue of showing `Deleted message` in LHN and other related issues are still relevant in the [proposal](https://github.com/Expensify/App/issues/23450#issuecomment-1658337106). How can we go about addressing them?

**Comment by jliexpensify** [comment]
Bump @rushatgabhane - some questions for you!

**Comment by rojiphil** [comment]
> @rojiphil could you please list the issues with repro steps, and expected results for all?

@rushatgabhane Looking into this today. Will share the details.

**Comment by rojiphil** [comment]
> @rojiphil could you please list the issues with repro steps, and expected results for all?

@rushatgabhane 
Please find the list of issues below. 

**Issue 1: Missing `[Deleted Message]`** 

Steps:
1) Send a message `Parent Message` in a Parent Report
2) `Reply in Thread` for the above message.
3) Send `Child Message` in Thread Report. Verify that LHN shows `Parent Message` as subtitle in Parent Report 
4) Delete `Parent Message` in Thread Report.
Actual Result: The subtitle for Parent Report in LHN displays a visible message previous to the `Parent Message`

Expected Result: The subtitle for Parent Report in LHN should display: `[Deleted message]`

<details>
<summary>Issue 1 Video</summary>

https://github.com/Expensify/App/assets/3004659/974573fc-964b-43f3-a38f-ab82f6ac0335


</details>

**Issue 2: [Deleted message] does not go away.**

Steps:
1) Let's have a Thread Report with `Parent Message` as report name and a child message `Child Message`
2) Delete the message `Child Message` in Thread Report. Verify that the LHN subtitle for Parent Report to this Thread Report still displays `Parent Message`
3) Delete the message `Parent Message` in Thread Report. Verify that the LHN subtitle for Parent Report shows `[Deleted message]`
4) Delete the last visible message in Parent Report. 

Actual Observation: The LHN subtitle for Parent Report keeps showing `[Deleted message]`. Further, deletion of messages in Parent Report will not change the subtitle and will keep showing `[Deleted message]`

Expected Observation: The LHN subtitle for Parent Report should show the immediately previous visible message.

<details>
<summary>Issue 2 Video</summary>

https://github.com/Expensify/App/assets/3004659/c2b34fda-32ac-40d4-b3c4-9acfa5007754

</details>

**Issue 3: Subtitle in LHN of Parent Report does not refresh when `Parent Message` is deleted from Thread Report.**

Note: This issue will only occur if we have fixed Issue 1.

Steps:
1) Let's have a Thread Report with `Parent Message` as report name without any child message 
2) Use preferences to go Offline
3) Delete `Parent Message` in Thread Report.

Actual Observation: The subtitle in LHN of Parent Report keeps showing 'Parent Message'

Expected Observation: The subtitle in LHN of Parent Report should show '[Deleted message]`

<details>
<summary>Issue 3 Video</summary>

https://github.com/Expensify/App/assets/3004659/204b208c-2134-4eb1-9791-640f58d74056

</details>

**Comment by rojiphil** [comment]
@rushatgabhane

Just thought to mention the link to the issues too if it can help.

> Issue 1: Missing `[Deleted Message]`

Issue 1 is #19722 (this one)

> Issue 2: [Deleted message] does not go away.

Issue 2 is #23450

> Issue 3: Subtitle in LHN of Parent Report does not refresh when `Parent Message` is deleted from Thread Report.

Issue 3 looks like another edge case

**Comment by chiragsalian** [comment]
As mentioned in the other issue. I realized that `isLastMessageDeletedParentAction` is no longer needed. The backend returns every parent action when fetching reports. And since the parent report action is the source of truth we should be able to use that to show `[Deleted Message]`.

The priority of this has slightly dropped since empty chats are not shown on the LHN unless explicitly navigated to but it would still be nice to patch this up.

**Comment by rojiphil** [comment]
@rushatgabhane Gentle bump for your review comments on issues [here](https://github.com/Expensify/App/issues/19722#issuecomment-1691484903) and [proposal here](https://github.com/Expensify/App/issues/23450#issuecomment-1658337106). Thanks

**Comment by jliexpensify** [comment]
Bump @rushatgabhane !

**Comment by trjExpensify** [comment]
I don't need to be on this issue as a second BZ, dropping off.

**Comment by rojiphil** [comment]
@rushatgabhane  FYI, I could reproduce [issue 3](https://github.com/Expensify/App/issues/19722#issuecomment-1691484903) mentioned in the context of [Money Request here in slack](https://expensify.slack.com/archives/C049HHMV9SM/p1693409176763969). When can we take this forward?

**Comment by jliexpensify** [comment]
Bumping @rushatgabhane for this [comment](https://github.com/Expensify/App/issues/19722#issuecomment-1702586720)

**Comment by jliexpensify** [comment]
Thanks @rushatgabhane - wow, this is a lengthy issue with a lot of discussion. It also [looks](https://github.com/Expensify/App/issues/19722#issuecomment-1692569211) like the priority of this has dropped with some back-end changes, and is now only a minor issue as indicated by @chiragsalian:

> The priority of this has slightly dropped since empty chats are not shown on the LHN unless explicitly navigated to but it would still be nice to patch this up.

So am I correct in assuming that we should now only focus on this issue, and source proposals for this specific scenario?

**Comment by rojiphil** [comment]
Have opened a discussion [here in slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1693976699584129) so that we can all come to same ground on this.

**Comment by jliexpensify** [comment]
Great, thank you @rojiphil !

**Comment by jliexpensify** [comment]
Looks like @rushatgabhane [you should be reviewing this](https://expensify.slack.com/archives/C01GTK53T8Q/p1693993646991529?thread_ts=1693976699.584129&cid=C01GTK53T8Q)?

**Comment by chiragsalian** [comment]
Not overdue, waiting on C+ review.

**Comment by rojiphil** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
LHN does not show `[Deleted message]` in Chat Report when the last visible action in Chat Report has a Thread with user comments but is also a deleted parent. 

### What is the root cause of that problem?

We are not setting the message `[Deleted message]` for deleted parent actions that have positive child visible action counts.

### What changes do you think we should make in order to solve the problem?

There are three changes that need to be made:

1) When `deleteReportComment` is called, we optimistically set the `lastMessageText` in the Report [here](https://github.com/Expensify/App/blob/e48121cb35052e958928c002d2b3a7d78a930ff3/src/libs/actions/Report.js#L956) by calling `getLastVisibleMessage`.  However, `getLastVisibleMessage` calls `getLastVisibleAction` which internally relies on `shouldReportActionBeVisibleAsLastAction
` as shown [here](https://github.com/Expensify/App/blob/c36ac0c98cf8a92a5b7136bbf090550cca5ea90c/src/libs/ReportActionsUtils.js#L386) to decide on the last action that should be considered as visible. However, as actions with empty text are ignored as [shown here](https://github.com/Expensify/App/blob/c36ac0c98cf8a92a5b7136bbf090550cca5ea90c/src/libs/ReportActionsUtils.js#L375) by checking for `isDeletedAction`, the last visible action for deleted parent action will not be considered. This will set the wrong `lastMessageText` in report.

To fix this problem, we can consider deleted parent actions with positive childVisibleActionCount as the last visible action by making the change like this [here](https://github.com/Expensify/App/blob/c36ac0c98cf8a92a5b7136bbf090550cca5ea90c/src/libs/ReportActionsUtils.js#L375)

```
        shouldReportActionBeVisible(reportAction, reportAction.reportActionID) && !(isWhisperAction(reportAction) && !isReportPreviewAction(reportAction)) && !(isDeletedAction(reportAction) && !isDeletedParentAction(reportAction))
```

2) So, now we can have the deleted parent action with positive childVisibleActionCount as the last visible action. But, then, the message text in `getLastVisibleMessage` [here](https://github.com/Expensify/App/blob/c36ac0c98cf8a92a5b7136bbf090550cca5ea90c/src/libs/ReportActionsUtils.js#L402) will still be empty although we expect `[Deleted message]` here. This change can be added [here](https://github.com/Expensify/App/blob/c36ac0c98cf8a92a5b7136bbf090550cca5ea90c/src/libs/ReportActionsUtils.js#L417) like this:

```
    if(isDeletedParentAction(lastVisibleAction) && isChatReport(reportID))
    {
        return {
            lastMessageText: Localize.translateLocal('parentReportAction.deletedMessage'),
        };
    }
```
3) One more thing is to consider is that the BE will update the `lastMessageText` with the previous action’s message. And, we use `getLastMessageTextForReport` as shown [here](https://github.com/Expensify/App/blob/c36ac0c98cf8a92a5b7136bbf090550cca5ea90c/src/libs/SidebarUtils.js#L276) to populate the LHN subtitle. So, to prevent the previous action’s message to be used when the last visible action is a deleted parent action with positive `childVisibleActionCount`, let us update the `getLastMessageTextForReport` to return `[Deleted message]`. The change can be made [at this location here](https://github.com/Expensify/App/blob/c36ac0c98cf8a92a5b7136bbf090550cca5ea90c/src/libs/OptionsListUtils.js#L403)

```
	else if (ReportActionUtils.isDeletedParentAction(lastReportAction) && ReportUtils.isChatReport(report)) {
	        lastMessageTextFromReport = Localize.translate(preferredLocale, 'parentReportAction.deletedMessage');          
   	 }

```
### What alternative solutions did you explore? (Optional)

**Comment by jliexpensify** [comment]
Hi @rushatgabhane - could you review the above proposal? Thanks!

**Comment by situchan** [comment]
This is not bug but expected. We don't show parent deleted message as last message text in LHN even though it has child visible actions in its thread.

**Comment by rojiphil** [comment]
> We don't show parent deleted message as last message text in LHN even though it has child visible actions in its thread.

@situchan Curious to know how we arrived at this conclusion and the reasoning behind it. Could you please point out the source of this conclusion?

**Comment by jliexpensify** [comment]
Bump @situchan - could you share the Slack link or GH comment please?

**Comment by situchan** [comment]
It was from https://github.com/Expensify/App/issues/22383#issuecomment-1715966986.
I raised [discussion](https://expensify.slack.com/archives/C01GTK53T8Q/p1695604109450219) again to confirm.

**Comment by rojiphil** [comment]
> @chiragsalian I like @rojiphil's [proposal](https://github.com/Expensify/App/issues/19722#issuecomment-1720740066)

@chiragsalian Gentle bump on taking this forward.

**Comment by chiragsalian** [comment]
just seeing this, your proposal looks good to me too. Feel free to create the PR @rojiphil

**Comment by rojiphil** [comment]
@rushatgabhane [PR](https://github.com/Expensify/App/pull/28364) is ready for review

**Comment by jliexpensify** [comment]
Payment Summary - this job is really messy so please weigh in and let me know if you have thoughts/anything's incorrect!

- Bug Reporter: @Nathan-Mulugeta $250
- Contributor: @rojiphil $1000
- C+: @rushatgabhane $1000 (New.Dot)

It looks like PR https://github.com/Expensify/App/pull/28364 took 14 days to merge - normally we'd cancel payment but it looks like even as the PR was happening, there was still a lot of debate. @chiragsalian what are you thoughts? Should we keep payment at $1000?

Also @dukenv0307 was assigned early onwards but didn't submit a PR to resolve this issue, right?

I will need to manually make a new Upworks job, so once we confirm the Payments, I'll sort that out.

**Comment by rojiphil** [comment]
> It looks like PR #28364 took 14 days to merge - normally we'd cancel payment but it looks like even as the PR was happening, there was still a lot of debate. @chiragsalian what are you thoughts? Should we keep payment at $1000?

@chiragsalian @rushatgabhane 

Considering that there was a gap of about 7 days ([this](https://github.com/Expensify/App/pull/28364#issuecomment-1760443357) minus [this](https://github.com/Expensify/App/pull/28364#issuecomment-1744748920)) when @rushatgabhane was not available to work on this issue, can we consider the total time taken to work on this issue as 7 days [[14](https://github.com/Expensify/App/issues/19722#issuecomment-1767066579) - 7(gap)]?

Further, there was one extended fix added [here](https://github.com/Expensify/App/pull/28364#issuecomment-1765569953) in the same PR. 
So, would it be reasonable to consider $1500 [i.e. $1000(this fix) + $500(extended fix)] as compensation here?

In addition, there is another extended fix mentioned [here](https://github.com/Expensify/App/pull/28364#issuecomment-1773769864). I am fine to include this extended fix too within the compensation of $1500 as mentioned above. 
Does this approach sound fair?

**Comment by chiragsalian** [comment]
> can we consider the total time taken to work on this issue as 7 days [https://github.com/Expensify/App/issues/19722#issuecomment-1767066579 - 7(gap)]?

Yeah I'm fine with that.

Yeah i think 1500 is fair for your work here.

**Comment by jliexpensify** [comment]
Great thanks @chiragsalian - will adjust the payments and make a new Upworks job.

**Comment by jliexpensify** [comment]
@chiragsalian just checking - do we apply the same logic for Rushat and give him $1500?

**Comment by Nathan-Mulugeta** [comment]
There is one other job open on upwork with this issue linked, it would be great if you close that one after processing payment to this issue as well, @jliexpensify

**Comment by jliexpensify** [comment]
Hmm can you link me the Upworks link @Nathan-Mulugeta ? I looked at the links in this GH and the jobs were all closed. Cheers!

**Comment by Nathan-Mulugeta** [comment]
I couldn't find a way to copy the link of an already hired contract from upwork.

**Comment by jliexpensify** [comment]
Is it a closed job though? Because what I'm finding is that Upworks randomly closes jobs after a month of inactivity (even if people are hired) which means we need to create new ones. There isn't an option to "re-open" a job.

**Comment by Nathan-Mulugeta** [comment]
It is not closed yet, it was open on Sep 27, so I think we can wait until it gets closed by upwork.

**Comment by rojiphil** [comment]
@jliexpensify 
Just checked in my Upwork account and noticed a duplicate offer for me. 
I have just sent a message via Upwork. 
You can close that one.

**Comment by rojiphil** [comment]
> @jliexpensify Just checked in my Upwork account and noticed a duplicate offer for me. I have just sent a message via Upwork. You can close that one.

And, here is the link if that helps
https://www.upwork.com/nx/wm/workroom/34966122/overview

![Screenshot 2023-10-24 at 11 46 05 AM](https://github.com/Expensify/App/assets/3004659/6e74eee4-406c-4d02-8aef-5ac69e81efdf)

**Comment by rojiphil** [comment]
> @jliexpensify Just checked in my Upwork account and noticed a duplicate offer for me. I have just sent a message via Upwork. You can close that one.

Here is the [link](https://github.com/Expensify/App/issues/19722#issuecomment-1737828617). Hope this helps

**Comment by jliexpensify** [comment]
Right @rojiphil and @Nathan-Mulugeta this is a closed job in Upworks:

![image](https://github.com/Expensify/App/assets/30815269/55471017-d8d7-46d9-b978-f18feb7b2c4d)

I also don't have an option to cancel these contracts, since it's closed:

![image](https://github.com/Expensify/App/assets/30815269/8697c3fe-f0a1-4806-93a1-a8db5caccba0)

**Comment by jliexpensify** [comment]
@Nathan-Mulugeta Can you help me understand what the issue is if there's an outstanding contract on a closed job? Does it affect your Upworks score or something?

You're the first Contributor who has reached out to me regarding this, so I want to understand what the issue is.

**Comment by jliexpensify** [comment]
Also bumping @rushatgabhane to complete the checklist!

**Comment by Nathan-Mulugeta** [comment]
Hey @jliexpensify the upwork Job that I was referring to is not the one you linked. I'll attach a screenshot of it below. But having an outstanding contract which upwork automatically closes might have an impact on the JSS (Job success score) of that account, but the impact might be low so we can ignore it. I just asked if it was possible for you to close it out before it manually gets closed. 

The job is this one:

![Screenshot_20231025-073654_Upwork Talent.png](https://github.com/Expensify/App/assets/111440031/2b0aed70-adbe-4787-8a2e-71499c7408a2)

**Comment by jliexpensify** [comment]
It looks like this is the Upworks job:

![image](https://github.com/Expensify/App/assets/30815269/824a724e-8b30-49ce-86f7-f27fffa373b0)

I'm a day ahead in Australia, so that's why it's showing as the 28th in the screenshot.

However, this job is closed, and I can't re-open it - that's why I had to make a new job. This seems like a super annoying quirk of Upworks, so I'll surface this with our team to see if we can disable the auto-close!

**Comment by jliexpensify** [comment]
Just circling back here @Nathan-Mulugeta and @situchan - I raised this with our team and someone discovered a workaround: anytime before the 4-week period is up, you can manually edit one letter in the Upworks description to extend the contract another 4 weeks. So Jack is going to look into a way to automatically edit these Upworks jobs before the 4 week period.

**Comment by jliexpensify** [comment]
[Paying now based off this summary.](https://github.com/Expensify/App/issues/19722#issuecomment-1773977956)

**Comment by jliexpensify** [comment]
Paid and job closed!

**Comment by rojiphil** [comment]
Received the payment. Thanks @jliexpensify
